### ISSUE TITLE
Re-introduce Scientific Notation

### ISSUE THREAD

**Issue by DelayedGaming** [SkriptLang/Skript#4553]
### Suggestion

Re-introduce scientific notation into Skript, which exists before 2.6 update. (i.e. `set {_x} to 1234 * 1e+6`)

### Why?

As mentioned above, scientific notation exists until Skript 2.6 update. I might have missed something of why this was removed after. Although it is not an important feature to have, but it is useful while performing calculations and such.

### Agreement

- [X] I have read the guidelines above and confirm I am following them with this suggestion.

**Comment by TPGamesNL** [comment]
Caused by #4275. The regex check doesn't take scientific notation (or [any other notation other than normal decimal](https://docs.oracle.com/javase/8/docs/api/java/lang/Double.html#valueOf-java.lang.String-)) into account. I don't recommend using the regex pattern in the link above, as it contains much stuff we don't need in Skript (e.g. hexadecimal, infinity, NaN).

Marking as enhancement since I don't think accepting scientific notation was ever intentional, given by the pattern in the docs:
![image](https://user-images.githubusercontent.com/29547183/151695549-57661260-febc-4b62-8a02-819db59a18ae.png)

**Comment by Moderocky** [comment]
I agree with TP, I don't think other notations are a necessary addition.
